Dev membership expiration reminder

zxzvipman opened the pull request:
Added a UI to modify when automated reminder emails are sent out

jjacobs10 committed:
Merge stable into expiration reminder (#43)

* Improved member search capabilities (#26)

* Improved search capabilties on manage users page. (#22)
Added searching by first name, last name, expiration date, and email. Fixed searching by tool authorization.

* Added membership heading feature

* Replaced if statement with ternary operator, removed magic number and uneccesary spaces.

* Added username column to the membership table (#29)

* Added username column to the member table.

* Updated member relationship dropdown, connected new_member_signup fields to the database.

* Updated while loop to be more readable.

* fixed the username parameters to start the numbers at 2

* add service space id for user, default space status: expired

* Updated membership heading with small edits based on feedback. (#34)

* Added pagination to the search member table. (#33)

* Removed commented out code

* Revert "Added pagination to the search member table. (#33)" (#36)

This reverts commit 187fb83be600fb6ec6d15d53c6cbec05697cc842.

* Revert "Merge branch 'dev-membership-expiration-reminder' into stable"

This reverts commit 050c9012833eebc8d02ebfecd94b9af97db3f95c, reversing
changes made to 8d42830e66bc43ed530711739fbf40736a64b079.

* Dev active membership heading (#38)

* Updated membership heading with small edits based on feedback.

* Updated color on membership heading for close to expiration date.

* Added a null check to the updated membership heading.

* Removed config.json and added to.gitignore

Co-authored-by: Austin Dobrusky <37594225+adobrusky@users.noreply.github.com>
Co-authored-by: Hadley Susie <hadleysusie123@gmail.com>
Co-authored-by: Hadley Susie <hsusie2@huskers.unl.edu>
Co-authored-by: Hadley Susie <69990293+hadleysusie@users.noreply.github.com>
Co-authored-by: Evan Mielak <evanmielak@gmail.com>
Co-authored-by: Evan Mielak <45115074+evanmielak@users.noreply.github.com>
Co-authored-by: adobrusky <austindobrusky@gmail.com>

jjacobs10 committed:
Finalized changes. Fixed some bugs

jjacobs10 committed:
Made Service Space ID parameter dynamic

taylor-runge committed:
Began work on UI for editing email expiration reminder peramiters.

taylor-runge committed:
Updated UI with success and error message as well as checking for some edge cases.

zxzvipman committed:
creating reminders model and connecting it with DB

connecting expiration's reminders table from the DB to the new model and modify the days before sending reminders according to DB.
connecting the UI to reminders model

taylor-runge committed:
Added check to ensure updated values are in the database as well as UI to display what the current values are to the user.

taylor-runge committed:
fixed spelling error

adobrusky commented on models/expiration_reminder.rb:
```diff
@@ -0,0 +1,12 @@
+require 'active_record'
+
+class ExpirationReminder < ActiveRecord::Base
+    def firstReminder
```
This is pretty minor but the rest of the Ruby code has methods and variables written in snake case and this is in camel case. Should probably conform to the rest of the code style and write these method names in snake case like `get_first_reminder` and `get_second_reminder`.

adobrusky commented on routes/admin/emails.rb:
```diff
@@ -12,6 +13,48 @@
+	reminder = ExpirationReminder.first
+
+	erb :'admin/manage_expiration_email', :layout => :fixed, :locals => {
+		:firstReminder => reminder.first_reminder,
```
Same as the above comment. The rest of the code writes the parameters in snake case so these should conform and be like `first_reminder` and `second_reminder`.

adobrusky reviewed (commented):
Besides these minor changes, the code looks good. I noticed while testing that if you enter a value for the first reminder, leave the second reminder empty, and save then the second reminder gets set to 0. If a field is left blank and you save I imagine it should not update the value that was blank in case the admins want to only modify one of the values at a time.

zxzvipman committed:
converting case type from camel case to snake case

changing firstReminder to get_first_reminder and changing secondReminder to get_second_reminder

taylor-runge committed:
Updated input for the email experation reminder prefrences to auto populate the input with the current values stored in the database.

adobrusky reviewed (approved):
Looks good
